doc/design.txt

```diff
@@ -210,15 +210,15 @@ An image trailer has the following structure:
      0                   1                   2                   3
```

Comment:
> You're doing permutation, while it would be more correct to consider it like a tree of states, if magic is unset do nothing, if magic is set check the other flag and so on and so forth.

I disagree with this assessment.

I'm enumerating all of the states and asking what the boot outcome would be in each one, which is a perfectly correct thing to do when considering "possible" states.

A decision tree representation is equivalent to a complete enumeration, which can be more compactly represented if many outcomes are equivalent due to some components being "don't care", such as your example of "if magic is unset do nothing".